azurelinuxagent/ga/monitor.py

```diff
@@ -137,29 +139,24 @@ def collect_and_send_events(self):
             self.protocol.report_event(event_list)
 
     def daemon(self, init_data=False):
+        try:
+            if init_data:
+                self.init_protocols()
+                self.init_imds_client()
+
+            while not self.stopped():
+                try:
+                    self.protocol.update_host_plugin_from_goal_state()
 
-        if init_data:
-            self.init_protocols()
-            self.init_imds_client()
-
-        min_delta = min(MonitorHandler.TELEMETRY_HEARTBEAT_PERIOD,
-                        MonitorHandler.CGROUP_TELEMETRY_POLLING_PERIOD,
-                        MonitorHandler.EVENT_COLLECTION_PERIOD,
-                        MonitorHandler.HOST_PLUGIN_HEARTBEAT_PERIOD,
-                        MonitorHandler.IMDS_HEARTBEAT_PERIOD).seconds
-        while not self.stopped():
-            try:
-                self.protocol.update_host_plugin_from_goal_state()
-                self._send_telemetry_heartbeat_op.run()
-                self._poll_telemetry_metrics_op.run()
-                self._collect_and_send_events_op.run()
-                self._send_host_plugin_heartbeat_op.run()
-                self._send_imds_heartbeat_op.run()
-                self._log_altered_network_configuration_op.run()
-                self._reset_loggers_op.run()
-            except Exception as e:
-                logger.warn("An error occurred in the monitor thread main loop; will skip the current iteration.\n{0}", ustr(e))
-            time.sleep(min_delta)
+                    for op in self._periodic_operations:
+                        op.run()
+
+                except Exception as e:
+                    logger.error("An error occurred in the monitor thread main loop; will skip the current iteration.\n{0}", ustr(e))
```

Comment:
Same for this, all functions have their own error handling or the PeriodicFunction.run will take care of it, this code path will never hit

Reply:
same as above

Reply:
BTW - the reason i added the try/except here is because sleep_until_next_operation in the finally was throwing on Python 2.6 (initially I was using timedelta.totalseconds() in it

Reply:
Ahhh that makes sense, finally can throw too didnt think of that lol. Thanks for adding it! :)